### PR TITLE
Parse features from schema file into Feature struct

### PR DESCRIPTION
**Pull request by pschmidt88**
This is a possible fix for issue #82

The features read from file will be parsed into a `map[string]api.Feature`, then
copied over to the `data` property of the `defaultStorage` struct.
This way they can be cast correctly to a feature when read in `repository.getToggle()`.

### CODE REVIEW

**Review by jrbarron — APPROVED** [review]
@pschmidt88 Thanks for your contribution! This seems like a reasonable fix. Ideally the `Storage` interface would be updated to use `map[string]api.Feature` as well, but I think that will have to wait until the next major release for compatibility reasons.

As to why this is like this, I can't really remember, but the original version of the library did not export the types from the `api` package so I suspect that is the reason. I'll wait for feedback from @FredrikOseberg before merging, but this seems like a reasonable fix for now. A unit test would perhaps be nice, but there wasn't one there to begin with so can't really fault you there 😆

**Review by FredrikOseberg — APPROVED** [review]
LGTM, thanks for this fix.

**Pull request merged**
